util/kube/ctl.go

```diff
@@ -63,14 +63,21 @@ func (k KubectlCmd) ApplyResource(config *rest.Config, obj *unstructured.Unstruc
 	if err != nil {
 		return "", err
 	}
-	applyArgs := []string{"--kubeconfig", f.Name(), "-n", namespace, "apply", "-f", "-"}
-	if dryRun {
-		applyArgs = append(applyArgs, "--dry-run")
+
+	var args []string
+	if needAuthReconcile(obj) {
```

Comment:
We should always do auth reconcile followed by auth apply.
Reconcile will ignore any non-role or rolebinding objects but does not set the previous config file. The subsequent apply is needed to set these fields so that subsequent applies will behave properly.